projects/front_power_distribution/src/lights_signal_fsm.c

```diff
@@ -0,0 +1,110 @@
+}
+
+FSM_STATE_TRANSITION(state_left_signal) {
+  SignalFsmStorage *storage = fsm->context;
+  FSM_ADD_GUARDED_TRANSITION(storage->signal_left_event, &prv_guard_off, state_none);
+  FSM_ADD_GUARDED_TRANSITION(storage->signal_right_event, &prv_guard_on, state_right_signal);
+  FSM_ADD_GUARDED_TRANSITION(storage->signal_hazard_event, &prv_guard_on, state_hazard_left_signal);
+}
+
+FSM_STATE_TRANSITION(state_right_signal) {
+  SignalFsmStorage *storage = fsm->context;
+  FSM_ADD_GUARDED_TRANSITION(storage->signal_left_event, &prv_guard_on, state_left_signal);
+  FSM_ADD_GUARDED_TRANSITION(storage->signal_right_event, &prv_guard_off, state_none);
+  FSM_ADD_GUARDED_TRANSITION(storage->signal_hazard_event, &prv_guard_on,
+                             state_hazard_right_signal);
+}
+
+FSM_STATE_TRANSITION(state_hazard_signal) {
+  SignalFsmStorage *storage = fsm->context;
+  FSM_ADD_GUARDED_TRANSITION(storage->signal_left_event, &prv_guard_on, state_hazard_left_signal);
+  FSM_ADD_GUARDED_TRANSITION(storage->signal_right_event, &prv_guard_on, state_hazard_right_signal);
+  FSM_ADD_GUARDED_TRANSITION(storage->signal_hazard_event, &prv_guard_off, state_none);
+}
+
+FSM_STATE_TRANSITION(state_hazard_left_signal) {
+  SignalFsmStorage *storage = fsm->context;
+  FSM_ADD_GUARDED_TRANSITION(storage->signal_left_event, &prv_guard_off, state_hazard_signal);
+  FSM_ADD_GUARDED_TRANSITION(storage->signal_right_event, &prv_guard_on, state_hazard_right_signal);
+  FSM_ADD_GUARDED_TRANSITION(storage->signal_hazard_event, &prv_guard_off, state_left_signal);
+}
+
+FSM_STATE_TRANSITION(state_hazard_right_signal) {
+  SignalFsmStorage *storage = fsm->context;
+  FSM_ADD_GUARDED_TRANSITION(storage->signal_left_event, &prv_guard_on, state_hazard_left_signal);
+  FSM_ADD_GUARDED_TRANSITION(storage->signal_right_event, &prv_guard_off, state_hazard_signal);
+  FSM_ADD_GUARDED_TRANSITION(storage->signal_hazard_event, &prv_guard_off, state_right_signal);
+}
+
+static void prv_state_none_output(Fsm *fsm, const Event *e, void *context) {
+  SignalFsmStorage *storage = context;
+  blink_event_generator_stop(&storage->blink_event_generator);
+}
+
+static void prv_state_left_signal_output(Fsm *fsm, const Event *e, void *context) {
+  SignalFsmStorage *storage = context;
+  blink_event_generator_start(&storage->blink_event_generator,
+                              FRONT_POWER_DISTRIBUTION_GPIO_EVENT_SIGNAL_LEFT);
+}
+
+static void prv_state_right_signal_output(Fsm *fsm, const Event *e, void *context) {
+  SignalFsmStorage *storage = context;
+  blink_event_generator_start(&storage->blink_event_generator,
+                              FRONT_POWER_DISTRIBUTION_GPIO_EVENT_SIGNAL_RIGHT);
+}
+
+static void prv_state_hazard_signal_output(Fsm *fsm, const Event *e, void *context) {
+  SignalFsmStorage *storage = context;
+  blink_event_generator_start(&storage->blink_event_generator,
+                              FRONT_POWER_DISTRIBUTION_GPIO_EVENT_SIGNAL_HAZARD);
```

Comment:
I actually want this module to be also a library, so take the input events: SIGNAL_*_EVENT and output events: GPIO_SIGNAL_*_EVENT as configuration...

Reply:
because both front and rear power distribution will use this :)